src/GameEngine/GameEngineDriver.cpp

```diff
@@ -149,5 +149,40 @@ namespace GameEngine {
             return true;
         }
 
+        bool test_tournament(){
+            Terminal::test_mode_on();
+            auto tournament_1 = new Tournament();
+            vector<int> input_vector = {2, 2, 7, 12, 2, 3, 3, 12};
+            Terminal::set_input_vector(input_vector);
+
+            // (2) when the tournament mode is chosen, the user is asked to select
+            tournament_1->prepareTournament(); // asks for num_maps, num_players, num_games, num_turns
+            Terminal::clear_terminal_input_counter();
+            // (3) after being started, the tournament runs without any user interaction
+            tournament_1->start();
+            // (4) upon completion, the results of the tournament are displayed as depicted above
+            tournament_1->displayResults();
+            //When finished, deleting everything
+            GameEngine::instance()->game_state->clear();
+
+            //assert( = );
```

Comment:
Commented out assert. Not completed as well. I don' t think your test is testing something right now

Reply:
@derkanadier

Reply:
This section has now been updated